### PR TITLE
SUS-3264 | introduce ListusersData::populateEventsLocalUsers

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/SUS-3264

Fills `specials.events_local_users` table with entries for a given wiki. This one replaces an old Perl backend script - `/usr/wikia/backend/bin/scribe/events_local_users.pl` - called by CreateNewWiki task.

Additionally:

* allow `Listusers` class to update `events_local_users` on devbox. This table is there.
* remove `LocalMaintenanceTask` class, seems to be a left-over after migration to Celery

### Example data

```
> (new ListusersData($wgCityId))->updateUserGroups( User::newFromName('Macbre'), User::newFromName('Macbre')->getGroups() )
```

```sql
mysql@geo-db-dev-db-slave.query.consul[specials]>select * from events_local_users where wiki_id = 5915;
+---------+---------+-----------+---------+-------+---------------------+---------------+------------+--------------+------------------+-----------------+----------------+
| wiki_id | user_id | user_name | last_ip | edits | editdate            | last_revision | cnt_groups | single_group | all_groups       | user_is_blocked | user_is_closed |
+---------+---------+-----------+---------+-------+---------------------+---------------+------------+--------------+------------------+-----------------+----------------+
|    5915 |  119245 |           |       0 | 13166 | 2017-11-21 11:57:50 |        114790 |          3 | util         | staff;sysop;util |               0 |              0 |
+---------+---------+-----------+---------+-------+---------------------+---------------+------------+--------------+------------------+-----------------+----------------+
1 row in set (0.00 sec)
```

### Queries

Get the list of all contributors for newly created wiki:

```sql
mysql@geo-db-g-slave.query.consul[devnotes]>select distinct(rev_user), rev_user_text from revision;
+----------+---------------+
| rev_user | rev_user_text |
+----------+---------------+
| 32769624 | FANDOM        |
| 32794352 | FANDOMbot     |
|  5275700 | BertH         |
| 33681148 | Tonda13       |
+----------+---------------+
4 rows in set (0.01 sec)
```

### CODE REVIEW

File: extensions/wikia/Listusers/SpecialListusers.php [hunk 1]
```diff
@@ -40,10 +40,7 @@
 $wgAjaxExportList[] = 'ListusersAjax::axShowUsers';
 $wgAjaxExportList[] = 'ListusersAjax::axSuggestUsers';
 
-// This tries to write to a database that the devboxes don't have write-permission for.
```
Comment (ghost) [inline extensions/wikia/Listusers/SpecialListusers.php:43]: :heart:

**Comment by WikiaTech** [comment]
SonarQube analysis reported 30 issues
* ![MAJOR][MAJOR] 22 major
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 5 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L85](https://github.com/Wikia/app/blob/d06795d733980d2a690e4ee81e470a46e50b0839/extensions/wikia/Listusers/SpecialListusers_helper.php#L85): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L98](https://github.com/Wikia/app/blob/d06795d733980d2a690e4ee81e470a46e50b0839/extensions/wikia/Listusers/SpecialListusers_helper.php#L98): The Cyclomatic Complexity of this function "loadData" is 26 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L98](https://github.com/Wikia/app/blob/d06795d733980d2a690e4ee81e470a46e50b0839/extensions/wikia/Listusers/SpecialListusers_helper.php#L98): This function "loadData" has 205 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L142](https://github.com/Wikia/app/blob/d06795d733980d2a690e4ee81e470a46e50b0839/extensions/wikia/Listusers/SpecialListusers_helper.php#L142): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L216](https://github.com/Wikia/app/blob/d06795d733980d2a690e4ee81e470a46e50b0839/extensions/wikia/Listusers/SpecialListusers_helper.php#L216): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L227](https://github.com/Wikia/app/blob/d06795d733980d2a690e4ee81e470a46e50b0839/extensions/wikia/Listusers/SpecialListusers_helper.php#L227): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L232](https://github.com/Wikia/app/blob/d06795d733980d2a690e4ee81e470a46e50b0839/extensions/wikia/Listusers/SpecialListusers_helper.php#L232): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L247](https://github.com/Wikia/app/blob/d06795d733980d2a690e4ee81e470a46e50b0839/extensions/wikia/Listusers/SpecialListusers_helper.php#L247): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L253](https://github.com/Wikia/app/blob/d06795d733980d2a690e4ee81e470a46e50b0839/extensions/wikia/Listusers/SpecialListusers_helper.php#L253): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L259](https://github.com/Wikia/app/blob/d06795d733980d2a690e4ee81e470a46e50b0839/extensions/wikia/Listusers/SpecialListusers_helper.php#L259): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by ghost — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 30 issues
* ![MAJOR][MAJOR] 22 major
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 5 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L85](https://github.com/Wikia/app/blob/6ffdcd3085f2004c3eb7cca74ba3ed76b53d9723/extensions/wikia/Listusers/SpecialListusers_helper.php#L85): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L98](https://github.com/Wikia/app/blob/6ffdcd3085f2004c3eb7cca74ba3ed76b53d9723/extensions/wikia/Listusers/SpecialListusers_helper.php#L98): The Cyclomatic Complexity of this function "loadData" is 26 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L98](https://github.com/Wikia/app/blob/6ffdcd3085f2004c3eb7cca74ba3ed76b53d9723/extensions/wikia/Listusers/SpecialListusers_helper.php#L98): This function "loadData" has 205 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L142](https://github.com/Wikia/app/blob/6ffdcd3085f2004c3eb7cca74ba3ed76b53d9723/extensions/wikia/Listusers/SpecialListusers_helper.php#L142): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L216](https://github.com/Wikia/app/blob/6ffdcd3085f2004c3eb7cca74ba3ed76b53d9723/extensions/wikia/Listusers/SpecialListusers_helper.php#L216): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L227](https://github.com/Wikia/app/blob/6ffdcd3085f2004c3eb7cca74ba3ed76b53d9723/extensions/wikia/Listusers/SpecialListusers_helper.php#L227): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L232](https://github.com/Wikia/app/blob/6ffdcd3085f2004c3eb7cca74ba3ed76b53d9723/extensions/wikia/Listusers/SpecialListusers_helper.php#L232): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L247](https://github.com/Wikia/app/blob/6ffdcd3085f2004c3eb7cca74ba3ed76b53d9723/extensions/wikia/Listusers/SpecialListusers_helper.php#L247): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L253](https://github.com/Wikia/app/blob/6ffdcd3085f2004c3eb7cca74ba3ed76b53d9723/extensions/wikia/Listusers/SpecialListusers_helper.php#L253): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L259](https://github.com/Wikia/app/blob/6ffdcd3085f2004c3eb7cca74ba3ed76b53d9723/extensions/wikia/Listusers/SpecialListusers_helper.php#L259): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by mszabo-wikia — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 56 issues
* ![MAJOR][MAJOR] 45 major
* ![MINOR][MINOR] 5 minor
* ![INFO][INFO] 6 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L85](https://github.com/Wikia/app/blob/15742146b52e0b622b11723de7b259f7ebae62b8/extensions/wikia/Listusers/SpecialListusers_helper.php#L85): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L98](https://github.com/Wikia/app/blob/15742146b52e0b622b11723de7b259f7ebae62b8/extensions/wikia/Listusers/SpecialListusers_helper.php#L98): The Cyclomatic Complexity of this function "loadData" is 26 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L98](https://github.com/Wikia/app/blob/15742146b52e0b622b11723de7b259f7ebae62b8/extensions/wikia/Listusers/SpecialListusers_helper.php#L98): This function "loadData" has 205 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L142](https://github.com/Wikia/app/blob/15742146b52e0b622b11723de7b259f7ebae62b8/extensions/wikia/Listusers/SpecialListusers_helper.php#L142): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L216](https://github.com/Wikia/app/blob/15742146b52e0b622b11723de7b259f7ebae62b8/extensions/wikia/Listusers/SpecialListusers_helper.php#L216): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L227](https://github.com/Wikia/app/blob/15742146b52e0b622b11723de7b259f7ebae62b8/extensions/wikia/Listusers/SpecialListusers_helper.php#L227): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L232](https://github.com/Wikia/app/blob/15742146b52e0b622b11723de7b259f7ebae62b8/extensions/wikia/Listusers/SpecialListusers_helper.php#L232): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L247](https://github.com/Wikia/app/blob/15742146b52e0b622b11723de7b259f7ebae62b8/extensions/wikia/Listusers/SpecialListusers_helper.php#L247): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L253](https://github.com/Wikia/app/blob/15742146b52e0b622b11723de7b259f7ebae62b8/extensions/wikia/Listusers/SpecialListusers_helper.php#L253): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [SpecialListusers_helper.php#L259](https://github.com/Wikia/app/blob/15742146b52e0b622b11723de7b259f7ebae62b8/extensions/wikia/Listusers/SpecialListusers_helper.php#L259): Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**